middlewares/middlewares.js

```diff
@@ -0,0 +1,49 @@
+const passport = require("passport");
+const { ExtractJwt, Strategy } = require("passport-jwt");
+const { config } = require("./config.js");
+const { User } = require("../models/user.model.js");
+const jwt = require("jsonwebtoken");
+
+const strategyOptions = {
+  secretOrKey: config.JWT_SECRET,
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+};
+
+passport.use(
+  new Strategy(strategyOptions, (payload, done) => {
+    User.findOne({ _id: payload.id })
+      .then((user) =>
+        !user ? done(new Error("User not existing")) : done(null, user)
+      )
+      .catch(done);
+  })
+);
+
+export const auth = (req, res, next) => {
+  passport.authenticate("jwt", { session: false }, (error, user) => {
+    if (!user || error)
```

Comment:
Tu też trzeba sprawdzić czy tokeny są takie same
```
  const [_bearer, token] = req.headers?.authorization?.split(' ');
  if (!user || error || !token || token !== user.token) {
    return res.status(401).json({ message: "Not authorized" });
  }
```

Reply:
*Dzień dobry, nie do końca rozumiem o co chodzi z samym _bearer ?

Reply:
jak robisz splita po spacji " " to wyjdzie ci tablica 2  wyrazów ten przedrostek bearere i właściwy token. I to jest zapis który odrazu tą tablice na zmienne rozdziela. A ta podłoga "_" jest dla tego że nie używamy tej zmiennej i czasem lsp może pokazywać bez tego _ ostrzerzenie o nieużywanej zmiennej.

Reply:
No właśnie pokazywał mi ten komunikat... dlatego się zastanawiałam, czy muszę go gdzieś wywołać...